### ISSUE TITLE
Support gradient accumulation

### ISSUE THREAD

**Issue by guillaumekln** [guillaumekln/OpenNMT-tf#2]
Either:

* via graph approach https://github.com/tensorflow/tensorflow/issues/27716
* via `tf.function`-based training loop (this requires a complete reimplementation of `Runner`, and also blocked by https://github.com/tensorflow/tensorflow/issues/29075)
